src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -0,0 +1,366 @@
+     * @param {string} id
+     * @param {TagsDescriptor} tagsDescriptor
+     * @returns {void}
+     */
+    public updateTagsById: (id: string, tagsDescriptor: TagsDescriptor) => void;
+
+    /**
+     * @name updateTagsForSelectedRegions
+     * @description uppdates tags for currently selected region with provided tags descriptor
+     * @param {TagsDescriptor} tagsDescriptor
+     * @returns {void}
+     */
+    public updateTagsForSelectedRegions: (tagsDescriptor: TagsDescriptor) => void;
+    private editor: Editor;
+    private primaryTag: Tag;
+    private secondaryTag: Tag;
+    private ternaryTag: Tag;
+    private incrementalRegionID: number;
+
+    constructor(props, context) {
+        super(props, context);
+
+        this.state = {
+            loaded: false,
+        };
+    }
+
+    public componentDidMount() {
+        const ct = CanvasTools;
+        const sz = document.getElementById("editorzone") as unknown as HTMLDivElement;
+        const tz = document.getElementById("toolbarzone")as unknown as HTMLDivElement;
+
+        // @ts-ignore
+        this.editor = new ct.Editor(sz);
+        this.editor.addToolbar(tz, ct.Editor.FullToolbarSet, "./../../../images/icons/");
+
+        // Expose CanvasTools Editor API
+        this.scaleRegionToFrameSize = this.editor.scaleRegionToFrameSize.bind(this.editor);
+        this.scaleRegionToSourceSize = this.editor.scaleRegionToSourceSize.bind(this.editor);
+
+        // Expose CanvasTools RegionManager API
+        this.addRegion = this.editor.RM.addRegion.bind(this.editor.RM);
+        this.addPointRegion = this.editor.RM.addPointRegion.bind(this.editor.RM);
+        this.addPolylineRegion = this.editor.RM.addPolylineRegion.bind(this.editor.RM);
+        this.addRectRegion = this.editor.RM.addRectRegion.bind(this.editor.RM);
+        this.deleteAllRegions = this.editor.RM.deleteAllRegions.bind(this.editor.RM);
+        this.deleteRegionById = this.editor.RM.deleteRegionById.bind(this.editor.RM);
+        this.freeze = this.editor.RM.freeze.bind(this.editor.RM);
+        this.getSelectedRegionsBounds = this.editor.RM.getSelectedRegionsBounds.bind(this.editor.RM);
+        this.redrawAllRegions = this.editor.RM.redrawAllRegions.bind(this.editor.RM);
+        this.resize = this.editor.RM.resize.bind(this.editor.RM);
+        this.selectRegionById = this.editor.RM.selectRegionById.bind(this.editor.RM);
+        this.toggleFreezeMode = this.editor.RM.toggleFreezeMode.bind(this.editor.RM);
+        this.unfreeze = this.editor.RM.unfreeze.bind(this.editor.RM);
+        this.updateTagsById = this.editor.RM.updateTagsById.bind(this.editor.RM);
+        this.updateTagsForSelectedRegions = this.editor.RM.updateTagsForSelectedRegions.bind(this.editor.RM);
+
+        this.incrementalRegionID = 100;
+
+        // Set up random tags for regions (will replace with footer interaction)
+        this.primaryTag = new ct.Core.Tag(
```

Comment:
Let's turn this into an array/map so we can support many tags on a region.

Reply:
This was placeholder before integrating with footer